treeherder/model/models.py

```diff
@@ -1451,16 +1499,55 @@ class TextLogError(models.Model):
     step = models.ForeignKey(TextLogStep, related_name='errors')
     line = models.TextField()
     line_number = models.PositiveIntegerField()
+    failure_line = models.OneToOneField(FailureLine,
+                                        related_name="text_log_error",
+                                        null=True)
+    # Note that the case of best_classification = None and best_is_verified = True
+    # has the special semantic that the line is ignored and should not be considered
+    # for future autoclassifications.
+    best_classification = models.ForeignKey("ClassifiedFailure",
+                                            related_name="best_for_errors",
+                                            null=True,
+                                            db_index=True,
+                                            on_delete=models.SET_NULL)
+    best_is_verified = models.BooleanField(default=False)
```

Comment:
I think it would be better if any mutable properties were stored in a seperate table /model as this will make subsequent modifications to the text log error table (if needed) a lot faster because we can use the copy table trick above. Also I suspect operations on the table/model would be faster (since it's a much smaller table).

I suspect a table like TextLogErrorClassification, with its primary key set to this one's, would work well. Similar to how I'm handling the taskcluster metadata in #2158.

I know this is late in the review cycle to point this out, and I apologize for that. It's only in the last week or so that some of these things have gelled in my head.